sherpa/astro/ui/utils.py

```diff
@@ -10422,48 +10437,45 @@ def get_source_plot(self, id=None, lo=None, hi=None, recalc=True):
 
         """
 
-        try:
-            d = self.get_data(id)
-        except IdentifierErr as ie:
-            if recalc:
-                raise ie
-            d = None
-
-        if isinstance(d, sherpa.astro.data.DataPHA):
-            plotobj = self._astrosourceplot
-            if recalc:
-                plotobj.prepare(d, self.get_source(id), lo=lo, hi=hi)
+        plotobj, data = self._get_plotobj("source", id=id, recalc=recalc)
+        if not recalc:
             return plotobj
 
-        return super().get_source_plot(id, recalc=recalc)
+        args = [data, self.get_source(id)]
+        kwargs = {}
+
+        if isinstance(data, sherpa.astro.data.DataPHA):
+            kwargs["lo"] = lo
+            kwargs["hi"] = hi
+
+        plotobj.prepare(*args, **kwargs)
+        return plotobj
 
     def get_fit_plot(self, id=None, recalc=True):
 
-        plotobj = self._fitplot
+        plotobj, data = self._get_plotobj("fit", id=id, recalc=recalc)
         if not recalc:
             return plotobj
 
-        d = self.get_data(id)
-        if isinstance(d, sherpa.astro.data.DataPHA):
+        if not isinstance(data, sherpa.astro.data.DataPHA):
```

Comment:
Perhaps we should repeat the logic of `get_fit_plot` from the superclass here, as we have done most of the work already?